### ISSUE TITLE
[no-use-before-define] `import * as React` marked as error in TSX/JSX file

### ISSUE THREAD

**Issue by DisappearedStar** [typescript-eslint/typescript-eslint#2502]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

.eslintrc.js
```js
{
    parser: '@typescript-eslint/parser',
    parserOptions: {
        sourceType: 'module',
        ecmaVersion: 2018,
        ecmaFeatures: {
            jsx: true,
        },
    },
    rules: {
        'no-use-before-define': 'off',
        '@typescript-eslint/no-use-before-define': ['error', {variables: false}],
    }
}
```
tsconfig.json
```json
{
  "compilerOptions": {
    "module": "esnext",
    "moduleResolution": "node",
    "target": "es5",
    "allowJs": true,
    "importHelpers": true,
    "jsx": "react",
    "experimentalDecorators": true,
    "isolatedModules": true,
    "importsNotUsedAsValues": "error"
  }
}
```

```TS
// foo.tsx

import * as React from 'react';

const App = () => (
    <span>hello</span>
);
```

**Expected Result**

No lint errors

**Actual Result**

`foo.tsx 1:13 React was used before it was defined. @typescript-eslint/no-use-before-define`

**Additional Info**

- AFAICT the error was introduced in 4.0.2-alpha.12 (alpha11 is ok)
- essentially, in the project I'm involved with every .tsx / .jsx file that contains React import is affected

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.0.2-alpha.15` |
| `@typescript-eslint/parser`        | `4.0.2-alpha.15` |
| `TypeScript`                       | `4.0.2` |
| `ESLint`                           | `7.8.1` |
| `node`                             | `10.16.3` |

**Comment by bradzacher** [comment]
Huh. That's an interesting side-effect. I hadn't thought about that rule doing that.

The usage is in exactly the same place as the definition, so it shouldn't be reporting. Seems like a bit of a bug in the rule that I will have to worm around.

**Comment by bradzacher** [comment]
I figured as much - the rule does `<` not `<=`, hence the false positive.
Easy fix.

**Comment by DisappearedStar** [comment]
Along with this issue, @typescript-eslint/parser 4.0.2-alpha.12 started to trigger `no-undef` at JSX intrinsic elements (i.e. `<div>`) in **.jsx** files.
I have read #2477 and corresponding [FAQ entry](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md#i-get-errors-from-the-no-undef-rule-about-global-variables-not-being-defined-even-though-there-are-no-typescript-errors). Adding `"JSX": "readonly"` global didn't help; adding each intrinsic element to globals would be very silly.

@bradzacher should I file a new issue?

**Comment by bradzacher** [comment]
nah don't worry. I was actually just thinking about that. I'll have a PR in a sec.

**Comment by DisappearedStar** [comment]
Just tried alpha18 — the abovementioned issues are gone. Thank you!

**Comment by nstrelow** [comment]
I have this as well, but for all normal react imports:

import React, { useState, useEffect } from "react";

and only when runnint eslint

**Comment by bradzacher** [comment]
https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md#i-am-using-a-rule-from-eslint-core-and-it-doesnt-work-correctly-with-typescript-code

You're using the wrong rule

**Comment by nstrelow** [comment]
Thanks a lot, wasn't aware of that. Will keep in mind in the future.

**Comment by gasscoelho** [comment]
> @bradzacher
> For me React imports are still broken on 4.0.2-alpha.20 and are working with 4.0.2-alpha.11
> 
> ```js
> import React, { useState, useEffect } from "react";
> ```
> 
> produces
> 
> ```
> error  'React' was used before it was defined  no-use-before-define
> ```
> 
> Did this change not make the alpha 20? Or somehow still broken?
> Thanks for your hard work 👍

Hi @nstrelow, did you find a solution for this? I'm also having the same message... should I disable this rule?

**Comment by bradzacher** [comment]
@gasscoelho did you actually read the thread...? I have already replied with the answer.

**Comment by gasscoelho** [comment]
I did but I didn't quite follow. I had a project configured on 3.10.1 which didn't have the [no-use-before-define] error on `import React from 'react'` and I had not disabled the rule. But today I was trying to configure a new project and I was having this warning on React import so that's why I asked if I should disable. Anyways, sorry for the misunderstanding. 🙂

**Comment by tibic** [comment]
"eslint-plugin-react": "^7.20.6",
"eslint": "^7.8.1",

error  'React' was used before it was defined  no-use-before-define

I dont know, feel confused

**Comment by Rennzie** [comment]
I'm also still getting this error with:  
`"@typescript-eslint/eslint-plugin": "^4.1.0"`
 `"@typescript-eslint/parser": "^4.1.0"`

**Comment by linonetwo** [comment]
> You're using the wrong rule

Hi confused guys, just use this in your config:

```
    'no-use-before-define': [0],
    '@typescript-eslint/no-use-before-define': [1],
```

Maybe it is just our preset don't aware of this rule yet, you can enable this new rule by hand: `'@typescript-eslint/no-use-before-define': [1],`

**Comment by bradzacher** [comment]
I'm going to go ahead and lock this to prevent further noise.

linonetwo's very helpful comments (thanks!) plus the FAQ and v4 release notes should be enough to solve this.

Please ensure you read them all, and understand that you need to manually update your config appropriately.
